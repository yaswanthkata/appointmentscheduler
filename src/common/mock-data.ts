import { Doctor } from "../types/Doctor";

export const MockDoctor: Doctor = {
  profile: {
    name: "LÃ©a Pereira",
    rating: 4,
    location: "Lisbon",
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam a mauris
    laoreet, accumsan magna sit amet, dapibus augue. Donec enim dolor,
    luctus in faucibus eu, viverra sit amet dolor. Proin eu viverra mi, ac
    consectetur nunc. Ut aliquet diam nec sem sodales, vitae hendrerit purus
    finibus. Nulla posuere ac neque sed vestibulum. Nullam et euismod risus.
    Nam ac dignissim justo. Vestibulum volutpat maximus orci nec ultricies.`,
    thumbnail:
      "https://images.unsplash.com/photo-1513721032312-6a18a42c8763?w=152&h=152&fit=crop&crop=faces",
    rate: "$150",
    specialization: "PSICOLOGIST"
  },
  availableDays: [
    {
      calendarDay: "2019-08-25",
      slots: [
        { time: "2019-08-25T08:00:00+0530", isAvailable: false },
        { time: "2019-08-25T08:30:00+0530", isAvailable: false },
        { time: "2019-08-25T09:00:00+0530", isAvailable: false },
        { time: "2019-08-25T09:30:00+0530", isAvailable: false },
        { time: "2019-08-25T10:00:00+0530", isAvailable: false }
      ]
    },
    {
      calendarDay: "2019-08-26",
      slots: [
        { time: "2019-08-26T08:00:00+0530", isAvailable: true },
        { time: "2019-08-26T08:30:00+0530", isAvailable: true },
        { time: "2019-08-26T09:00:00+0530", isAvailable: true },
        { time: "2019-08-26T09:30:00+0530", isAvailable: true },
        { time: "2019-08-26T10:00:00+0530", isAvailable: true }
      ]
    },
    {
      calendarDay: "2019-08-27",
      slots: [
        { time: "2019-08-27T08:00:00+0530", isAvailable: true },
        { time: "2019-08-27T08:30:00+0530", isAvailable: true },
        { time: "2019-08-27T09:00:00+0530", isAvailable: true },
        { time: "2019-08-27T09:30:00+0530", isAvailable: true },
        { time: "2019-08-27T10:00:00+0530", isAvailable: true }
      ]
    },
    {
      calendarDay: "2019-08-28",
      slots: [
        { time: "2019-08-28T08:00:00+0530", isAvailable: true },
        { time: "2019-08-28T08:30:00+0530", isAvailable: true },
        { time: "2019-08-28T09:00:00+0530", isAvailable: true },
        { time: "2019-08-28T09:30:00+0530", isAvailable: true },
        { time: "2019-08-28T10:00:00+0530", isAvailable: true }
      ]
    },
    {
      calendarDay: "2019-08-29",
      slots: [
        { time: "2019-08-29T08:00:00+0530", isAvailable: true },
        { time: "2019-08-29T08:30:00+0530", isAvailable: true },
        { time: "2019-08-29T09:00:00+0530", isAvailable: true },
        { time: "2019-08-29T09:30:00+0530", isAvailable: true },
        { time: "2019-08-29T10:00:00+0530", isAvailable: true }
      ]
    }
  ]
};
