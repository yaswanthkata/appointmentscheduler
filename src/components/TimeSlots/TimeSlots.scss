.time-slots {
  text-align: center;
  margin-top: 1em;
  .time-slot {
    cursor: pointer;
    color: #858585;
    line-height: 1;
    margin-bottom: 0.5em;
    display: flex;
    align-items: center;
    justify-content: center;
    user-select: none;
  }
  .no-slot {
    font-size: large;
    padding: 11px;
    cursor: auto;
  }
}
