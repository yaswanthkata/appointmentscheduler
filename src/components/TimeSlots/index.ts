export {default as TimeSlots} from './TimeSlots';