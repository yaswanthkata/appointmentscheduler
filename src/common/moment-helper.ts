import moment, { Moment } from "moment";
import { MomentSlot, CalendarDay } from "../types";
import { MockDoctor } from "./mock-data";

export const getDaySlots = (day: Moment): MomentSlot[] => {
  let convertedSlots: MomentSlot[] = [];
  const availableDay = MockDoctor.availableDays.find(
    d => day.format("YYYY-MM-DD") === d.calendarDay
  );
  if (availableDay) {
    convertedSlots = availableDay.slots.map(slot => {
      return {
        time: moment(slot.time),
        isAvailable:slot.isAvailable
      };
    });
  }
  return convertedSlots;
};

export const getFirstAvailableStartDate = (dates: CalendarDay[]) => {
  const moments: Moment[] = [];
  dates.forEach(d => moments.push(moment(d.calendarDay, "YYYY-MM-DD")));
  return moment.min(moments);
};
