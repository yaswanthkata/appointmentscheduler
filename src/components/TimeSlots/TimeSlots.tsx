import * as React from "react";
import { MomentSlot } from "../../types";
import "./TimeSlots.scss";

export interface ITimeSlotsProps {
  slots: MomentSlot[];
  onSlotClick(selectedSlot: MomentSlot): React.MouseEventHandler<any>;
}

export default function TimeSlots({ slots, onSlotClick }: ITimeSlotsProps) {
  return (
    <div className="time-slots">
      {slots.map(slot => (
        <div key={slot.time.toString()} className="time-slot">
          {slot.isAvailable ? (
            <button className="btn" onClick={onSlotClick(slot)}>
              {slot.time.format("HH:mm")}
            </button>
          ) : (
            <div className="no-slot">-</div>
          )}
        </div>
      ))}
    </div>
  );
}
