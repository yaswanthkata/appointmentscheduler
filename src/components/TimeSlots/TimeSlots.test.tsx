import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import TimeSlots, { ITimeSlotsProps } from "./TimeSlots";
import moment, { Moment } from "moment";
import * as MomentHelper from "../../common/moment-helper";

const props: ITimeSlotsProps = {
  slots: MomentHelper.getDaySlots(moment()),
  onSlotClick: jest.fn()
};

let componentWrapper: ShallowWrapper<React.Component>;

beforeEach(() => {
  componentWrapper = shallow(<TimeSlots {...props} />);
});

it("should render Timeslots component correctly", () => {
  expect(componentWrapper).toMatchSnapshot();
});

it("should render  a list of slots", () => {
  const slots = componentWrapper.find("button");
  expect(slots.length).toEqual(5);
});

it("should call onSlotClick on slot item click", () => {
  componentWrapper
    .find(".btn")
    .at(0)
    .simulate("click");
  expect(props.onSlotClick).toHaveBeenCalled();
});
