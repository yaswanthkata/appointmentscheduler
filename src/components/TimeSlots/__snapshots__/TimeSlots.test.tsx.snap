// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Timeslots component correctly 1`] = `ShallowWrapper {}`;
